bridges/otelzap/encoder.go

```diff
@@ -196,15 +194,13 @@ type arrayEncoder struct {
 }
 
 func (a *arrayEncoder) AppendArray(v zapcore.ArrayMarshaler) error {
-	// TODO: Use arrayEncoder from a pool.
 	arr := &arrayEncoder{}
```

Comment:
Wouldn't pre-allocating the slice in `arrayEncoder` reduce the amount of heap allocations (similarly to `newObjectEncoder(2)` for object encoder)?

This can be handled in a separate PR.